rand_core/src/lib.rs

```diff
@@ -91,15 +91,18 @@ pub mod le;
 /// It is recommended that implementations also implement:
 /// 
 /// - `Debug` with a custom implementation which *does not* print any internal
-///   state (at least, [`CryptoRng`]s should not risk leaking state through Debug)
+///   state (at least, [`CryptoRng`]s should not risk leaking state through
+///   `Debug`).
 /// - `Serialize` and `Deserialize` (from Serde), preferably making Serde
-///   support optional at the crate level in PRNG libs
-/// - `Clone` if, and only if, the clone will have identical output to the
-///   original (i.e. all deterministic PRNGs but not external generators)
-/// - *never* implement `Copy` (accidental copies may cause repeated values)
-/// - also *do not* implement `Default`, but instead implement `SeedableRng`
-///   thus allowing use of `rand::NewRng` (which is automatically implemented)
-/// - `Eq` and `PartialEq` could be implemented, but are probably not useful
+///   support optional at the crate level in PRNG libs.
+/// - `Clone`, but only if the clone will have identical output to the original
+///   (i.e. be a true clone), or if the clone would be another handle to the
+///   same generator.
```

Comment:
I never really got the point of saying this. If you put *anything* in an `Rc<..>` type then `clone()` will copy the reference, not the object — so why even say it?

If you're going to be pedantic like this then you might as well update the next point about `Copy` too, though I don't see much point.

Reply:
I copied it from your comment, but now understand why you left it out. With `OsRng` in mind I think it starts to make more sense, as that is the same principle but without `Rc<..>`?

Reply:
Sort of. I'm wondering if we should say much about `Clone` at all now. Maybe just that on normal PRNGs `clone` should imply identical output? Even that may not make sense given that we might try to make `StdRng` reseed on clone in the future(?).

Reply:
I just reworded it as "`Clone` if possible"